### ISSUE TITLE
vmadm destroy doesn't tell logadm to stop rotating logs

### ISSUE THREAD

**Issue by anttisyn** [joyent/smartos-live#319]
After destroying two vm's I got this in email every hour.

logadm: Warning: /zones/a550c00b-a107-42ee-80b6-6809f04d5494/root/tmp/vm.log.6: No such file or directory
logadm: Warning: /zones/a550c00b-a107-42ee-80b6-6809f04d5494/root/tmp/vm.log.5: No such file or directory
logadm: Warning: /zones/a550c00b-a107-42ee-80b6-6809f04d5494/root/tmp/vm.log.4: No such file or directory
logadm: Warning: /zones/419187bb-a4d5-496e-8aac-11bec292bad9/root/tmp/vm.log.3: No such file or directory
logadm: Warning: /zones/a550c00b-a107-42ee-80b6-6809f04d5494/root/tmp/vm.log.3: No such file or directory
logadm: Warning: /zones/419187bb-a4d5-496e-8aac-11bec292bad9/root/tmp/vm.log.2: No such file or directory
logadm: Warning: /zones/a550c00b-a107-42ee-80b6-6809f04d5494/root/tmp/vm.log.2: No such file or directory
logadm: Warning: /zones/419187bb-a4d5-496e-8aac-11bec292bad9/root/tmp/vm.log.1: No such file or directory
logadm: Warning: /zones/a550c00b-a107-42ee-80b6-6809f04d5494/root/tmp/vm.log.1: No such file or directory
logadm: Warning: /zones/419187bb-a4d5-496e-8aac-11bec292bad9/root/tmp/vm.log.0: No such file or directory
logadm: Warning: /zones/a550c00b-a107-42ee-80b6-6809f04d5494/root/tmp/vm.log.0: No such file or directory
logadm: Warning: /zones/419187bb-a4d5-496e-8aac-11bec292bad9/root/tmp/vm.log: No such file or directory
logadm: Warning: /zones/a550c00b-a107-42ee-80b6-6809f04d5494/root/tmp/vm.log: No such file or directory

SmartOS Live Image v0.147+ build: 20140418T031241Z

Workaround was to use logadm -r to remove the files from rotation.

**Comment by RSully** [comment]
I've been digging into this bug ([OS-3097](https://smartos.org/bugview/OS-3097), as mentioned above). 

I wanted to post a bit of a brain dump. This could be completely wrong as I have only skimmed a very tiny portion of the logadm codebase, but it sounds like it could be on the right track.

/cc @rmustacc @joshwilsdon 

***

logadm gets executed in 2 primary spots from SmartOS: [crontab](https://github.com/joyent/smartos-live/blob/be5fc42716a719f5cb60a9559cea470beea0b7eb/overlay/generic/etc/cron.d/crontabs/root#L4) and [rotate-kvm-logs.sh](https://github.com/joyent/smartos-live/blob/be5fc42716a719f5cb60a9559cea470beea0b7eb/src/vm/sbin/rotate-kvm-logs.sh#L72-L78) (which is actually executed from [crontab](https://github.com/joyent/smartos-live/blob/be5fc42716a719f5cb60a9559cea470beea0b7eb/overlay/generic/etc/cron.d/crontabs/root#L6) too).

From my *limited* understanding, logadm writes to the timestamps file regardless of how it is invoked. It does this because it needs to keep track of the last time any path was rotated and needs to support cases where the logname is different than the log file name and probably when there are wildcards, etc.. This change may be relevant: illumos/illumos-gate@34969e746608c8170ab540c876836f625f25d8eb.

The problem would appear that logadm is executing on entries from the timestamps file as if they were requests in the config file.

I would also propose that KVM VMs should remove the logadm timestamp entry when a VM is deleted. You could take that a step further and remove the need for rotate-kvm-logs.sh completely by properly registering the logs with logadm in the first place. Alternatively, logadm could delete the entry from the timestamps file if the log file no longer exists.

With that said, it sounds like OS-3097 may be overly prescriptive, and if fixed blindly could lead to its own set of problems.

**Comment by naisanza** [comment]
@bahamat thanks!

**Comment by bassu** [comment]
If you are like me and launch and destroy **hundreds** of zones, here's a quick one liner (as the bug is still lurking around):

```
for v in $(logadm -V |  awk -F/ '/zones/ {print $3}'); do exist=$(vmadm lookup uuid=$v); \
 [[ -z "$exist" ]] && logadm -r /zones/$v/root/tmp/vm.log; done
```

**Comment by badboy** [comment]
Any chance this will be addressed?

**Comment by sjorge** [comment]
Was testing with some kvm instances today, It's rather annoying! @melloc ?

**Comment by badboy** [comment]
*bump*

**Comment by omnigaz** [comment]
bump

**Comment by jcea** [comment]
Suffering this too. BUMP!

**Comment by bassu** [comment]
Bump * 150920^39

**Comment by zssfred** [comment]
same here(20161013T025521Z ). The dirty-n-quick way is manually removing the destroyed entry in /var/logadm/timestamps.

**Comment by jurajlutter** [comment]
BUMP

**Comment by TonyBostonTB** [comment]
still

**Comment by melloc** [comment]
Fixed in 1d55b897d3, which will be in this week's upcoming release.
